internal/pkg/skuba/deployments/ssh/cri.go

```diff
@@ -17,11 +17,52 @@
 
 package ssh
 
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/SUSE/skuba/pkg/skuba"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
 func init() {
+	stateMap["cri.configure"] = criConfigure
 	stateMap["cri.start"] = criStart
+	stateMap["cri.reset"] = crioReset
+}
+
+func criConfigure(t *Target, data interface{}) error {
+	criFiles, err := ioutil.ReadDir(skuba.CriDir())
+	if err != nil {
+		return errors.Wrap(err, "Could not read local cri directory: "+skuba.CriDir())
+	}
+	defer t.ssh("rm -rf /tmp/cri.d")
+
+	for _, f := range criFiles {
+		if err := t.target.UploadFile(filepath.Join(skuba.CriDir(), f.Name()), filepath.Join("/tmp/cri.d", f.Name())); err != nil {
```

Comment:
why are you uploading file to a temporary directory? what is that `/tmp` dir used for? I only see you are copying `/tmp/cri.d/default_flags` from that directory...

Reply:
The same technique is used for other configurations within skuba:

- cni.go `	if err := t.target.UploadFile(filepath.Join(skuba.CniDir(), f.Name()), filepath.Join("/tmp/cni.d", f.Name())); err != nil {`

- dex.go `if err := t.target.UploadFile(filepath.Join(skuba.DexDir(), f.Name()), filepath.Join("/tmp/dex.d", f.Name())); err != nil {`

- gangway.go `	if err := t.target.UploadFile(filepath.Join(skuba.GangwayDir(), f.Name()), filepath.Join("/tmp/gangway.d", f.Name())); err != nil {`

- kubeadm.go `remoteKubeadmUpgradeConfFile := filepath.Join("/tmp/", skuba.KubeadmUpgradeConfFile())`

- kured.go `if err := t.target.UploadFile(filepath.Join(skuba.KuredDir(), f.Name()), filepath.Join("/tmp/kured.d", f.Name())); err != nil {`

- psp.go `	if err := t.target.UploadFile(filepath.Join(skuba.PspDir(), f.Name()), filepath.Join("/tmp/psp.d", f.Name())); err != nil {`

So for the sake of consistency, I followed the same pattern. This temporary directory gets deleted upon the completion of the state.

Reply:
We would refactor this, but in another PR.

Reply:
Ok, I see...